lua/mapvote/sv_mapvote.lua

```diff
@@ -87,127 +87,74 @@ local function mapVoteOver( autoGamemode, callback )
-                    if string.match( map, pattern ) then
-                        gamemode = gm.name
-                        break
-                    end
-                end
-            end
-        end
-    else
-        print( "not enabled" )
-    end
-
     timer.Simple( 4, function()
-        if ( hook.Run( "MapVoteChange", map ) ~= false ) then
-            if ( callback ) then
-                callback( map )
-            else
-                -- if map requires another gamemode then switch to it
-                if ( gamemode and gamemode ~= engine.ActiveGamemode() ) then
-                    RunConsoleCommand( "gamemode", gamemode )
-                end
-                RunConsoleCommand( "changelevel", map )
-            end
+        if hook.Run("MapVoteChange", map) == false then return end
+        if callback then
+            callback( map )
         end
+
+        RunConsoleCommand( "changelevel", map )
     end )
 end
 
-function MapVote.Start( length, current, limit, prefix, callback )
-    current = current or MapVote.Config.AllowCurrentMap or false
-    length = length or MapVote.Config.TimeLimit or 28
-    limit = limit or MapVote.Config.MapLimit or 24
-    prefix = prefix or MapVote.Config.MapPrefixes
-    local cooldown = MapVote.Config.EnableCooldown or MapVote.Config.EnableCooldown == nil and true
-    local autoGamemode = autoGamemode or MapVote.Config.AutoGamemode or MapVote.Config.AutoGamemode == nil and true
+local function isMapAllowed(m)
+    local conf = MapVote.Config
+    local prefixes = conf.MapPrefixes
+    if prefixes and type(prefixes) == "string" then -- This should be done at configuration step
+        prefixes = { prefixes }
+    end
 
-    local is_expression = false
+    if not MapVote.AllowCurrentMap and m == game.GetMap() then return false end -- dont allow current map in vote
+    if MapVote.Config.EnableCooldown == true and table.HasValue( recentmaps, m ) then return false end -- dont allow recent maps in vote
+    if MapVote.Config.ExcludedMaps[m] then return false end -- dont allow excluded maps in vote
 
-    if not prefix then
-        local info = file.Read( GAMEMODE.Folder .. "/" .. GAMEMODE.FolderName .. ".txt", "GAME" )
+    if MapVote.Config.IncludedMaps[m] then return true end -- skip prefix check if map is in included maps
 
-        if info then
-            info = util.KeyValuesToTable( info )
-            prefix = info.maps
-        else
-            error( "MapVote Prefix can not be loaded from gamemode" )
+    for _, v in pairs(prefixes) do
+        if string.find( m, "^" .. v ) then
+            return true
         end
-
-        is_expression = true
-    else
-        if prefix and type( prefix ) ~= "table" then prefix = { prefix } end
     end
+    return false
+end
 
-    local maps = file.Find( "maps/*.bsp", "GAME" )
+function MapVote.Start( length, callback )
+    length = length or MapVote.Config.TimeLimit or 28
```

Comment:
not sure why we should keep this 28, might be nicer to have the fallback to a whole number like 30? turbo nitpick ofc

Reply:
that was the default before